midp/gfx.js

```diff
@@ -278,10 +268,27 @@ var currentlyFocusedTextEditor;
         converterFunc(rgbData, abgrData, width, height, 0, width);
 
         context.putImageData(ctxImageData, 0, 0);
-
-        setImageData(imageData, width, height, context);
     };
 
+    function setImageDataDimensions(imageData, w, h) {
+        imageData.klass.classInfo.getField("I.width.I").set(imageData, w);
+        imageData.context.canvas.width = w;
+
+        imageData.klass.classInfo.getField("I.height.I").set(imageData, h);
+        imageData.context.canvas.height = h;
+    }
+
+    Override["javax/microedition/lcdui/ImageData.<init>.()V"] = function() {
+        this.context = document.createElement("canvas").getContext("2d");
```

Comment:
The ImageData class that the FavouriteArtists MIDlet is using is not javax/microedition/lcdui/ImageData but a custom class that they've written (it's a private class defined in the Control.java file itself).

The ImageData constructor has package visibility, so it's only accessible to classes in the javax/microedition/lcdui package (and the only class that creates ImageData objects is ImageDataFactory).